.github/workflows/release.yml

```diff
@@ -0,0 +1,38 @@
+name: Publish
```

Comment:
Ideally I'd like the CI setup to mirror other pydantic projects for simplicity.

Best to have one file called `ci.yml`, [`ci.yml` from pytest-pretty](https://github.com/samuelcolvin/pytest-pretty/blob/main/.github/workflows/ci.yml) is the latest iteration of that format, so probably a good candidate for copying.